packaging/dependency/lxd.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package dependency
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/juju/packaging"
+)
+
+const blankSeries = ""
+
+// LXD is a dependency instance for installing lxd.
+var LXD packaging.Dependency = lxdDependency{}
+
+type lxdDependency struct{}
+
+// PackageList implements packaging.Dependency.
+func (lxdDependency) PackageList(series string) ([]packaging.Package, error) {
+	var pkg packaging.Package
+
+	switch series {
+	case "centos7", "opensuseleap":
+		return nil, errors.NotImplementedf("installing lxd on %s", series)
```

Comment:
With all the abstractions around what package managers we use, we don't actually install anything but Ubuntu? (at least I can't track an install path from Dependency to, eg, Zypper). How is that?

Reply:
Given that the recommended install procedure for lxd on centos7 is basically "install snap and snap install lxd" I would be surprised if that ever worked for non-ubuntu machines...

So just in case we ever do reach this code in the future (I think you can do that if you add-machine --series centos7 and juju add-machine lxd:X) I added the not implemented bit so the error makes more sense to the user.

Reply:
I'm asking, "are there any packages we *actually* install from a package manager outside of Ubuntu"? All the Dependencies that you implemented seem to only support Ubuntu, were there others that you just didn't get to yet?

Reply:
We install series/os-specific packages in the cloud-init stuff (e.g. https://github.com/juju/juju/blob/a5ab92ec9b7f5da3678d9ac603fe52d45af24412/cloudconfig/cloudinit/cloudinit_centos.go#L34). We could create a dependency for those as well.